SyntheticGeometry/Affine.lagda.md

```diff
@@ -0,0 +1,207 @@
+        (λ ((A , coupled-A) , SpecA≡X)
+        → let A≡X→k : A ≡ (pointwiseAlgebra X kₐ)
+              A≡X→k =
+                A                            ≡⟨ uaCommAlgebra ((_ , coupled-A) , snd (canonical-hom A)) ⟩
+                pointwiseAlgebra (Spec A) kₐ  ≡⟨ cong (λ u → pointwiseAlgebra u kₐ) SpecA≡X ⟩
+                pointwiseAlgebra X kₐ ∎
+          in subst (λ u → ⟨ is-coupled-algebra u ⟩) A≡X→k coupled-A)
+        X∈ImSpec
+    Iso.rightInv SpecIso (X , X∈ImSpec) =
+      Σ≡Prop (λ _ → isPropPropTrunc)
+             (sym (ua (to-ev-hom X , X-coupled)))
+      where
+      X-coupled : ⟨ is-coupled X ⟩
+      X-coupled =
+        PT.rec
+          (snd (is-coupled X))
+          (λ{ ((A , coupled-A) , SpecA≡X) → subst (fst ∘ is-coupled) SpecA≡X (is-equiv-to-ev-hom A coupled-A)})
+          X∈ImSpec
+    Iso.leftInv SpecIso (A , coupled-A) =
+      Σ≡Prop (λ B → snd (is-coupled-algebra B))
+             (sym (uaCommAlgebra (((to-ev-map A) , coupled-A) , (snd (canonical-hom A)))))
+
+Spec-coupled-embedding : isEmbedding (Spec-coupled {ℓ' = ℓ})
+Spec-coupled-embedding = isEmbeddingFromIsEquivToImage Spec-coupled Spec-equiv-onto-image
+
+```
+
+From this we can conclude a couple of things, most notably,
+that the Σ-type of all coupled algebras with spectrum equal to a given type is a proposition.
+
+```agda
+
+Spec-path-equiv : {A B : CommAlgebra k ℓ}
+  → ⟨ is-coupled-algebra A ⟩ → ⟨ is-coupled-algebra B ⟩
+  → (A ≡ B) ≃ (Spec A ≡ Spec B)
+Spec-path-equiv {A = A} {B = B} coupled-A coupled-B =
+  A ≡ B                              ≃⟨ Σ≡PropEquiv (λ A → snd (is-coupled-algebra A)) ⟩
+  (A , coupled-A) ≡ (B , coupled-B)  ≃⟨ _ , Spec-coupled-embedding (A , coupled-A) (B , coupled-B) ⟩
+  Spec A ≡ Spec B ■
+
+Spec-path-equiv-fp : {A B : CommAlgebra k ℓ}
+  → isFPAlgebra A → isFPAlgebra B
+  → (A ≡ B) ≃ (Spec A ≡ Spec B)
+Spec-path-equiv-fp {A = A} {B = B} fp-A fp-B = Spec-path-equiv (k-sqc A fp-A) (k-sqc B fp-B)
+
+coupled-affine-is-prop : (X : Type ℓ)
+  → isProp (Σ[ (A , coupled-A) ∈ (Σ[ A ∈ CommAlgebra k ℓ ] ⟨ is-coupled-algebra A ⟩ ) ] Spec A ≡ X)
+coupled-affine-is-prop X = isEmbedding→hasPropFibers Spec-coupled-embedding X
+
+fp-affine-is-prop : (X : Type ℓ)
+  → isProp (Σ[ (A , fp-A) ∈ (Σ[ A ∈ CommAlgebra k ℓ ] isFPAlgebra A) ] Spec A ≡ X)
+fp-affine-is-prop X ((A , fp-A) , SpecA≡X) ((B , fp-B) , SpecB≡X) i = (replaceFst i) , (cong snd fromCoupled i)
+  where
+    coupled-A = k-sqc A fp-A
+    coupled-B = k-sqc B fp-B
+
+    fromCoupled : ((A , coupled-A) , SpecA≡X) ≡ ((B , coupled-B) , SpecB≡X)
+    fromCoupled = coupled-affine-is-prop X ((A , coupled-A) , SpecA≡X) ((B , coupled-B) , SpecB≡X)
+
+    replaceFst : (A , fp-A) ≡ (B , fp-B)
+    replaceFst = Σ≡Prop (λ _ → isPropPropTrunc) (cong (fst ∘ fst) fromCoupled)
+
+```
+
+Finite affine qc-open covers
+----------------------------
+The following is used to define qc-schemes:
+
+```agda
+
+is-affine-finite-qc-open-cover : {n : ℕ}
```

Comment:
I feel like this should be in `qc-Scheme.lagda.md`. Are there reasons to have it here?

Reply:
Moved